api/http/handler/stacks/stack_delete.go

```diff
@@ -245,7 +245,21 @@ func (handler *Handler) deleteStack(userID portainer.UserID, stack *portainer.St
 		}
 
 		out, err := handler.KubernetesDeployer.Remove(userID, endpoint, manifestFiles, stack.Namespace)
-
+		if err != nil {
+			// If the error pertains to a missing manifest file, we can safely ignore it.
+			// This allows us to clean up relevant records in the stack, such as the stack
+			// database entry and the /compose/stack folder. This suppression is safe
+			// because before the current API (/stacks/{id} [delete]) is called, another
+			// API has already removed the Kubernetes application deployments/pods by
+			// sending the delete request (/apis/apps/v1/namespaces/default/deployments/xx)
+			// to the Kubernetes cluster API endpoint via the Portainer endpoint proxy.
+			// See @https://github.com/portainer/portainer/blob/release/2.20/app/kubernetes/views/applications/applicationsController.js#L96
+			for _, manifest := range manifestFiles {
+				if _, statErr := os.Stat(manifest); os.IsNotExist(statErr) {
+					return nil
+				}
+			}
+		}
```

Comment:
are we sure about it? basically, this code says - if removal has failed and one of the manifest files is missing, we can consider this stack as removed. Is this correct?

if it is correct, I think we should try to reduce comments to one line. even something like `if removal has failed and one of the manifest files is missing, we can consider this stack as removed` is enough.

writing multiline comments adds complexity to the code, can get out of sync fast (what if one of the apis or controllers you've mentioned changes?), and it's hard to see if the comment is still true, as we need to read it and understand the code it's commenting.